Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGToDBAnalyzer.cs

```diff
@@ -0,0 +1,163 @@
+            }
+
+            var accessor = KeyValueStoreAccessor.Open(storeDirectory: OutputDirPath);
+
+            if (accessor.Succeeded)
+            {
+                Accessor = accessor.Result;
+                m_accessorSucceeded = true;
+            }
+            else
+            {
+                Console.WriteLine("Could not access RocksDB datastore. Exiting analyzer.");
+            }
+        }
+
+        /// <inheritdoc/>
+        public override int Analyze()
+        {
+            if (!m_accessorSucceeded)
+            {
+                return 0;
+            }
+
+            Analysis.IgnoreResult(
+              Accessor.Use(database =>
+              {
+                  foreach (var invEvent in m_invocationEventList.BXLInvEventList)
+                  {
+                      var eq = new EventTypeQuery
+                      {
+                          EventTypeID = (int)ExecutionEventId.DominoInvocation,
+                          UUID = invEvent.UUID
+                      };
+
+                      database.Put(eq.ToByteArray(), invEvent.ToByteArray());
+                  }
+              })
+            );
+
+            Accessor.Dispose();
+            return 0;
+        }
+
+        /// <inheritdoc/>
+        public override bool CanHandleEvent(ExecutionEventId eventId, uint workerId, long timestamp, int eventPayloadSize)
+        {
+            if (eventId.Equals(ExecutionEventId.DominoInvocation))
+            {
+                Console.WriteLine("Found a valid domino invocation event. The worker id is {0}, the timestamp is {1}, and the payload size is {2}.", workerId, timestamp, eventPayloadSize);
+                WorkerID = workerId; // store workerID to pass into protobuf object to identify this event
+
+                return true;
+            }
+            return false; // return false to keep the event from being parsed
+        }
+
+        /// <summary>
+        /// Override the DominoInvocationEvent to capture its data and store it in the protobuf 
+        /// </summary>
+        public override void DominoInvocation(DominoInvocationEventData data)
+        {
+            var domInvEvent = new BXLInvocationEvent();
+            var loggingConfig = data.Configuration.Logging;
+
+            var uuid = Guid.NewGuid().ToString();
+
+            domInvEvent.UUID = uuid;
+            domInvEvent.WorkerID = WorkerID;
+            domInvEvent.SubstSource = loggingConfig.SubstSource.ToString(PathTable, PathFormat.HostOs);
+            domInvEvent.SubstTarget = loggingConfig.SubstTarget.ToString(PathTable, PathFormat.HostOs);
+            domInvEvent.IsSubstSourceValid = loggingConfig.SubstSource.IsValid;
+            domInvEvent.IsSubstTargetValid = loggingConfig.SubstTarget.IsValid;
+
+            m_invocationEventList.BXLInvEventList.Add(domInvEvent);

```

Comment:
>m_invocationEventList [](start = 12, length = 21)

Why not store this directly to the DB as events from the XLG are read in instead of holding collecting them in memory?

Reply:
I think I will add this in a future PR but this is a good point for scalability (perhaps a sliding window of XYZ # of records in order to do some batch writing to rocksDB). 

---
In reply to: [306409428](https://github.com/microsoft/BuildXL/pull/636#discussion_r306409428) [](ancestors = 306409428)